### PR TITLE
PIM-7349: fix empty family when using quick export of products

### PR DESCRIPTION
**Pull request by ahocquard**
When exporting products with quick export, some family were null.

Actually, this is the same fix as in PIM-7183. It is a Doctrine bug with Proxy objects being null after a clearing of the Entity Manager. 
See  https://github.com/akeneo/pim-community-dev/pull/7641.

The (dirty) fix is to change the way to iterate the objects.
I fixed it in all the reader.

It was already done for one reader (for an SLA), but not for all.
https://github.com/akeneo/pim-community-dev/pull/7565/files

But I think it was badly fixed and I think it's possible to have an infinite loop.
Also, I deleted the two behats from this SLA. They are useless, creating 103 products (!), and not business oriented at all. 

**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-2.0.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 ## Bug fixes
 
 - PIM-7323: Fix unique data synchronizer to handle value removal
+- PIM-7349: Fix empty family when using quick export of products
+
+# 2.0.x
```
Comment (juliensnz) [inline CHANGELOG-2.0.md]: this line is missplaced

File: src/Pim/Bundle/EnrichBundle/spec/Connector/Reader/MassEdit/FilteredProductAndProductModelReaderSpec.php [hunk 2]
```diff
@@ -111,7 +111,7 @@ function () use (&$productsCount) {
             }
         );
         $cursor->current()->will(new ReturnPromise($products));
-        $cursor->next()->shouldBeCalledTimes(4);
+        $cursor->next()->shouldBeCalledTimes(5);
```
Comment (ahocquard) [inline src/Pim/Bundle/EnrichBundle/spec/Connector/Reader/MassEdit/FilteredProductAndProductModelReaderSpec.php]: the spec was wrong: we have 6 products and product models, so it's normal to have `next` 5 times.

File: features/mass-action/mass_edit_many_products_at_once.feature [hunk 3]
```diff
@@ -1,56 +0,0 @@
-@javascript
```
Comment (juliensnz) [inline features/mass-action/mass_edit_many_products_at_once.feature]: why did you removed this?

File: src/Pim/Bundle/EnrichBundle/Connector/Reader/MassEdit/FilteredProductAndProductModelReader.php [hunk 4]
```diff
@@ -191,6 +193,8 @@ private function getNextProduct(): ?ProductInterface
             if (!$this->firstRead) {
                 $this->productsAndProductModels->next();
             }
+
+            $this->firstRead = false;
```
Comment (ahocquard) [inline src/Pim/Bundle/EnrichBundle/Connector/Reader/MassEdit/FilteredProductAndProductModelReader.php]: avoid infinite loop

File: features/mass-action/mass_edit_many_products_at_once.feature (hunk 3 above)
Comment (ahocquard) [reply features/mass-action/mass_edit_many_products_at_once.feature]: please read the description

File: CHANGELOG-2.0.md (hunk 1 above)
Comment (ahocquard) [reply CHANGELOG-2.0.md]: changelog rebase...

**Pull request merged**